src/twisted/internet/endpoints.py

```diff
@@ -747,7 +751,7 @@ def ebLog(error):
         d.addCallback(cbDeliver)
         d.addErrback(ebLog)
         d.addBoth(lambda ignored: resolutionReceiver.resolutionComplete())
-        return resolutionReceiver
+        return resolution
```

Comment:
This changes the return value here, but it is never used:

https://github.com/twisted/twisted/blob/6a8a70b348480cc59e9944f47330ea0f4a6934f7/src/twisted/internet/endpoints.py#L959-L961

Reply:
Hm... might not be used by Twisted own code, but this looks like public API, so it might be used by someone else.

But I agree that the change makes more sense... since the old code just returns an input argument.

But this needs to follow the deprecations process... or ask for an exception 

-------------

Do we still need `@rtype: L{IResolutionReceiver}` in the docstring ?

same for `@type resolutionReceiver: L{IResolutionReceiver}` ?

Reply:
> Hm... might not be used by Twisted own code, but this looks like public API, so it might be used by someone else.
> 
> But I agree that the change makes more sense... since the old code just returns an input argument.

I don't think I was clear above -- this implementation (i.e. `_nameResolver` which gets set from a private class also in this file) returns the wrong type, which is never used. I don't see anyway for this to leak out to the public API. This is an internal-only change.